image/actions.sh

```diff
@@ -51,6 +51,13 @@ function detect-tfmask() {
   export TFMASK
 }
 
+function execute_run_commands() {
+  if [[ -n $INPUT_RUN ]]; then
+    echo "Executing init commands specified in 'run' parameter"
+    eval "$INPUT_RUN"
```

Comment:
This should be run in subshell.

Reply:
you got an example of what you mean?

Reply:
`eval` will execute in the current process, so local variables and shell settings are exposed to the INPUT_RUN script. It could even change them and break things. If we run it in a subshell e.g. `echo $($INPUT_RUN)` then it's run a new shell process that can't change our environment.

Thinking about it, for best isolation lets just write the script to a file and execute that, e.g.

```bash
printf "%s" "$INPUT_RUN" > /.input_run.sh
bash /.input_run,sh
```